<launch>
    <!-- Launch Rosserial Communication-->
    <node name="serial_node" pkg="rosserial_python" type="serial_node.py" output="screen">
        <param name="port" value="/dev/ttyAMC0" />
        <param name="baud" value="115200" />
    </node>

    <!-- Launch Lidar-->
    <node name="ydlidar_node"  pkg="ydlidar"  type="ydlidar_node" output="screen" respawn="false" >
        <param name="port"         type="string" value="/dev/ydlidar"/>  
        <param name="baudrate"     type="int"    value="115200"/>
        <param name="frame_id"     type="string" value="laser"/>
        <param name="low_exposure"  type="bool"   value="false"/>
        <param name="resolution_fixed"    type="bool"   value="true"/>
        <param name="auto_reconnect"    type="bool"   value="true"/>
        <param name="reversion"    type="bool"   value="false"/>
        <param name="angle_min"    type="double" value="-180" />
        <param name="angle_max"    type="double" value="180" />
        <param name="range_min"    type="double" value="0.12" />
        <param name="range_max"    type="double" value="16.0" />
        <param name="ignore_array" type="string" value="" />
        <param name="samp_rate"    type="int"    value="9"/>
        <param name="frequency"    type="double" value="7"/>
    </node>

    <!-- Static transform from laser to base-->
    <node pkg="tf"
        type="static_transform_publisher"
        name="laser2base"
        args="0 0 0.1 0 3.14159 0 base_link laser 10" />

</launch>
